app/scripts/proactive/view/dom.js

```diff
@@ -1037,43 +1045,90 @@ define(
             $.getScript("studio-conf.js", function () {
                 $("#documentationLinkId").attr("href", config.docUrl);
             });
-
+            $('#workflow-designer-outer').on('contextmenu', function(e) {
+                    var top = e.offsetY + 30;
+                    var left = e.offsetX - 10;
+                    $(".context-menu-task").hide();
+                    $(".context-menu-canvas").css({
+                      top: top,
+                      left: left
+                    }).show();
+                    return false; //blocks default Webbrowser right click menu
+                  }).on("click", function() {
+                     $(".context-menu-canvas").hide();
+                   });
+            $(".context-menu-canvas li").on("click", function() {
+                                 $(".context-menu-canvas").hide();
+                               });
 
             var ctrlDown = false;
-            var ctrlKey = 17, commandKey = 91, vKey = 86, cKey = 67, zKey = 90, yKey = 89, aKey = 65;
+            var ctrlKey = 17, commandKey = 91, vKey = 86, cKey = 67, zKey = 90, yKey = 89, aKey = 65, xKey = 88;
             var pasteAllow = true;
             var canDoPast = false
             $('#workflow-designer-outer').bind('keydown', function (e) {
                 if (e.keyCode == ctrlKey || e.keyCode == commandKey) ctrlDown = true;
             }).keyup(function (e) {
                 if (e.keyCode == ctrlKey || e.keyCode == commandKey) ctrlDown = false;
             });
+            function copyTasks(){
+                copiedTasks = [];
+                positions = [];
+                console.log("copy");
+                $(".selected-task").each(function (i, t) {
+                positions.push({left: $(t).position().left, top: $(t).position().top})
+                    copiedTasks.push($(t).data( "view" ))
+                })
+                // let the user how he can do past(ctr-v)
+                if(copiedTasks.length > 0){
+                    StudioClient.alert('Copy/Paste', 'Click on the canvas where you want to paste and then do ctrl-V.', 'warning');
+                } else {
+                    StudioClient.alert('Copy/Paste', 'Select at least one task.', 'warning');
+                }
+
+                console.log('positions')
+                console.log(positions)
```

Comment:
Not sure that those log are required and if you thinks that they are required should put them in one line.